Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -326,7 +326,12 @@ protected override void Reload()
 
                 base.Reload();
 
-                SyncFileSystem();
+                if (this.IsShowingAllFiles)
+                {
+                    // We only need to sync the files on load if we're actually showing them
+                    // otherwise we'll sync on idle.
+                    this.SyncFileSystem();
```

Comment:
Does this code get re-executed if you toggle show all files on/off? I do that frequently working in NTVS projects.

Reply:
As discussed, no. Show all files enumerates the files we know about, which is a collection that is constructed on idle. I'll ensure this collection is correct when toggling show all files. In the next PR, when we no longer do the disk enumeration on the UI thread.